packages/Caching/Config/FileHashComputer.php

```diff
@@ -17,7 +18,7 @@ public function compute(string $filePath): string
         $this->ensureIsPhp($filePath);
 
         $parametersHash = SimpleParameterProvider::hash();
-        return sha1($filePath . $parametersHash);
+        return sha1(FileSystem::read($filePath) . $parametersHash);
```

Comment:
another solution is by using `sha1_file()` instead:

```php
        return sha1_file($filePath);
```

when `rector.php` changed, even only space, the hash also changed:

```bash
➜  rector-src git:(ensure-always-refresh) php -r 'echo sha1_file(__DIR__ . "/rector.php");';
71902587bce991de196cf2c1f79f1083e3834bbf%      

// change the file ...
// re-run
                                                                                                                                 
➜  rector-src git:(ensure-always-refresh) php -r 'echo sha1_file(__DIR__ . "/rector.php");';
59b7d5b0e9d9672186774efb691a9ccc8dd5ebac% 
```

Reply:
@TomasVotruba @staabm what do you think?

Reply:
also, `sets` is not part of `$parametersHash`, so only rely on  `$parametersHash`, change `$rectorConfig->sets()` content won't change the cached files, so definitely requires 2 options:

- read the file + parameter hash like current proposed solution
- use sha1_file() instead.

on possible improvement: when using sets, I think the existing rules inside sets need to be extracted + hashed, then compare so when new rule added into set, the cache automatically cleared, but that may need a different PR :)

Reply:
> sha1_file() instead

👍

---


What comes to mind: should we also support auto cache clear when imported configs change?

Reply:
as I commented previously, for imported config changed, eg: via sets() method, that should be in separate PR.

Rector version changed should be also in separate PR to ease review and revert.

Reply:
I updated to use `sha1_file()` https://github.com/rectorphp/rector-src/pull/4899/commits/7a3b7b34d91b5095cb137790720ab3473cc74f18